app/models/participant_profile.rb

```diff
@@ -57,4 +61,10 @@ def validation_decision(name)
     decision = validation_decisions.find { |record| record.validation_step.to_s == name.to_s }
     decision || validation_decisions.build(validation_step: name)
   end
+
+  def sync_teacher_profile
+    self.teacher_profile = user.teacher_profile || user.build_teacher_profile
```

Comment:
What is `build_teacher_profile`?

Reply:
It is a `belongs_to` equivalent of `records.build`: https://guides.rubyonrails.org/association_basics.html#belongs-to-association-reference